extensions/interactions/CodeRepl/directives/oppia-interactive-code-repl.component.ts

```diff
@@ -0,0 +1,292 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the CodeRepl interaction.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { AfterViewInit, ChangeDetectorRef, Component, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { CodemirrorComponent } from '@ctrl/ngx-codemirror';
+import { Subscription } from 'rxjs';
+
+import { CodeReplCustomizationArgs } from 'interactions/customization-args-defs';
+import { InteractionAttributesExtractorService } from 'interactions/interaction-attributes-extractor.service';
+import { CurrentInteractionService, InteractionRulesService } from 'pages/exploration-player-page/services/current-interaction.service';
+import { PlayerPositionService } from 'pages/exploration-player-page/services/player-position.service';
+import { CodeReplRulesService } from './code-repl-rules.service';
+
+@Component({
+  selector: 'oppia-interactive-code-repl',
+  templateUrl: './code-repl-interaction.component.html',
+  styleUrls: []
+})
+export class InteractiveCodeReplComponent implements
+  OnInit, AfterViewInit, OnDestroy {
```

Comment:
```suggestion
    OnInit, AfterViewInit, OnDestroy {
```

Reply:
When it is intended like this I firstly thought these are some inputs.

Reply:
Oh... thanks for pointing it out, done!